service/controller/clusterapi/v29/resource/tccpsecuritygroupid/create.go

```diff
@@ -0,0 +1,69 @@
+package tccpsecuritygroupid
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v29/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v29/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := r.toClusterFunc(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var groups []*ec2.SecurityGroup
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding ingress security groups")
+
+		i := &ec2.DescribeSecurityGroupsInput{
+			Filters: []*ec2.Filter{
+				{
+					Name: aws.String("tag:Name"),
+					Values: []*string{
+						aws.String(key.SecurityGroupName(cr, "ingress")),
```

Comment:
This needs only the cluster ID, which can be gathered using the `LabelsGetter` interface, so a pointer to the machine deployment CR is enough.